wrapper/main/src/main/kotlin/org/quiltmc/qkl/wrapper/main/registry/RegistryDsl.kt

```diff
@@ -0,0 +1,121 @@
+package org.quiltmc.qkl.wrapper.main.registry
+
+import net.minecraft.util.Identifier
+import net.minecraft.util.registry.Registry
+
+/**
+ * A RegistryObject with a name parameter, registers under your modid
+ * @param modid The modid to register under
+ * @param name The name to register under
+ * @param t The object to register
+ *
+ * @author Oliver-makes-code (Emma)
+ * */
+public data class RegistryObject<T>(val modid: String, val name: String, val t: T) {
+    /**
+     * Registers a RegistryObject
+     * @param registry The registry to register to
+     *
+     * @author Oliver-makes-code (Emma)
+     * */
+    public infix fun toRegistry(registry: Registry<T>): T {
+        return Registry.register(registry, Identifier(modid, name), this.t)
+    }
+}
+
+/**
+ * Represent an action with a registry
+ * @param modid The modid to register under
+ * @param registry The registry to register to
+ *
+ * @author Oliver-makes-code (Emma)
+ * */
+public data class RegistryAction<T>(val modid: String?, val registry: Registry<T>) {
+    /**
+     * Registers an object with a given Identifier
+     * @param id The Identifier to register under
+     *
+     * @author Oliver-makes-code (Emma)
+     * */
+    public infix fun T.withId(id: Identifier) {
+        Registry.register(registry, id, this)
+    }
+
+    /**
+     * Registers an object with a given name
```

Comment:
Probably worth adding its behavior with/without a modid, i.e make it clear how the `name`  param is used